mitodl/mitxonline
# Search: adds additional fields to the courses API

#### Pre-Flight checklist


#### What are the relevant tickets?

Fixes #857 

#### What's this PR do?

Adds `description`, `current_price`, and `page_url` to the `courses` API:
- `description` is an HTML-stripped copy of what's in the description field in the course's associated page
- `current_price` is the price from the _most expensive_ product assigned to the next most recent course run. (It uses `first_unexpired_run` to determine this.)
- `page_url` is the URL to the course's page. This is a relative URL.
- `page` is the `CoursePageSerializer` data for the course's page. (This includes the Flexible Pricing form URL and the featured image; may be useful.) 

#### How should this be manually tested?

Automated tests should pass.

Load either the list of courses or a specific course using the API. (For example, `http://mitxonline.odl.local:8013/api/courses/2/`.) You should see `page`, `page_url`, `description`, and `current_price` populated accordingly. If there are no products for the next course run (`next_run_id`), the price will be null; similarly, if there's no course page, the other 3 fields will be null. 

Changing the page for a given course or modifying the product for the course run should be reflected in the API results.
